libafl/src/inputs/encoded.rs

```diff
@@ -96,28 +96,40 @@ where
 }
 
 impl InputDecoder for TokenInputEncoderDecoder {
-    fn decode(&mut self, input: &EncodedInput, bytes: &mut Vec<u8>) -> Result<(), Error> {
-        // Use a specific rand state each decode, to make sure we decode it in the same way
-        // But try to mix up the seed, as it empirically yields better results
-        let codes = input.codes();
-        let mut rand = StdRand::new();
-        let mut prev_len = 0;
-        for id in codes {
-            let tok = self
-                .id_table
-                .get(&(id % self.next_id))
-                .ok_or_else(|| Error::illegal_state(format!("Id {id} not in the decoder table")))?;
-            if self.encoding_type == TokenizationKind::WithWhitespace {
-                // If we encode whitespace, mutations can f*ck up validity
-                // by removing whitespace or single char tokens like `+` to
-                // create illegal entries like Variable1"String". We need to
-                // fix such entries. So wherever we have two tokens that have
-                // a length larger than 1 and are not whitespace we randomly
-                // insert whitespace or a single byte token from our token vec.
-                // We do this randomly because this enhances the overall
-                // mutation quality.
+    fn decode(&mut self, input: &mut EncodedInput, bytes: &mut Vec<u8>) -> Result<(), Error> {
+        let codes = &mut input.codes;
+        if self.encoding_type == TokenizationKind::WithWhitespace {
+            // If we encode whitespace, mutations can f*ck up validity
+            // by removing whitespace or single char tokens like `+` to
+            // create illegal entries like Variable1"String". We need to
+            // fix such entries. So wherever we have two tokens that have
+            // a length larger than 1 and are not whitespace, we randomly
+            // insert whitespace or a single byte token from our token vec.
+            // We do this randomly because this enhances the overall
+            // mutation quality.
+            let mut positions: Vec<usize> = vec![];
+            let mut prev_len = 0;
+
+            for pos in 0..codes.len() {
+                let tok = self
+                    .id_table
+                    .get(&(codes[pos] % self.next_id))
+                    .ok_or_else(|| {
+                        Error::illegal_state(format!("Id {} not in the decoder table", codes[pos]))
+                    })?;
                 let len = tok.len();
                 if prev_len > 1 && len > 1 {
+                    positions.push(pos);
+                }
+                prev_len = len;
+            }
+
+            if positions.len() > 0 {
+                // We need to repair the input
+                let mut rand = StdRand::new();
```

Comment:
This won't work on no_std.
Maybe use my earlier rand seed, based on the contents/len of `codes` in this case?

Reply:
is this correct now for nostd?